web-services/client/src/main/java/datawave/webservice/query/QueryParametersImpl.java

```diff
@@ -89,11 +110,21 @@ public void validate(Map<String,List<String>> parameters) throws IllegalArgument
                 throw new IllegalArgumentException("Unknown condition.");
             }
         }
-        Preconditions.checkNotNull(this.query, "Query string cannot be null");
-        Preconditions.checkNotNull(this.queryName, "Query name cannot be null");
-        Preconditions.checkNotNull(this.persistenceMode, "Persistence mode cannot be null");
-        Preconditions.checkNotNull(this.auths, "Query auths cannot be null");
-        Preconditions.checkNotNull(this.expirationDate, "Expiration date cannot be null");
+        
+        try {
+            Preconditions.checkNotNull(this.query, "QueryParameter 'query' cannot be null");
+            Preconditions.checkNotNull(this.queryName, "QueryParameter 'queryName' cannot be null");
+            Preconditions.checkNotNull(this.persistenceMode, "QueryParameter 'persistence' mode cannot be null");
+            Preconditions.checkNotNull(this.auths, "QueryParameter 'auths' cannot be null");
+            Preconditions.checkNotNull(this.pagesize, "QueryParameter 'pagesize' cannot be null");
+            Preconditions.checkNotNull(this.pageTimeout, "QueryParameter 'pageTimeout' cannot be null");
+            Preconditions.checkNotNull(this.expirationDate, "QueryParameter 'expiration' date cannot be null");
+            Preconditions.checkNotNull(this.beginDate, "QueryParameter 'begin' cannot be null");
+            Preconditions.checkNotNull(this.endDate, "QueryParameter 'end' cannot be null");
+            Preconditions.checkNotNull(this.logicName, "QueryParameter 'logicName' cannot be null");
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage());
```

Comment:
You might keep the "caused by" exception here. I know the message will tell us the variable but it doesn't hurt to have the original line number of the error as well: `throw new IllegalArgumentException("Missing required parameters", e)`

Reply:
Makes a lot of sense, I'll add that in